modules/xfem/include/userobjects/CrackMeshCut3DUserObject.h

```diff
@@ -66,31 +113,50 @@ class MeshCut3DUserObject : public GeometricCutUserObject
   /// Number of steps to grow the mesh
   unsigned int _n_step_growth;
 
-  /// variables to help control the work flow
+  /// Variables to help control the work flow
   bool _stop;
   bool _grow;
 
   /// Boundary nodes of the cutter mesh
   std::vector<dof_id_type> _boundary;
 
+  /// Active boundary nodes where growth is allowed
+  std::vector<std::vector<dof_id_type>> _active_boundary;
+
+  /// Inactive boundary
+  std::vector<unsigned int> _inactive_boundary_pos;
+
+  /// Front nodes that are grown from the crack front definition defined in the input
+  /// therefore, they are (1) in the same order as defined in the input and (2) the number of nodes does not change
+  std::vector<dof_id_type> _tracked_crack_front_points;
```

Comment:
I think we could use a sideset on the cutting mesh for this purpose, rather than requiring that node IDs be specified.